### PR TITLE
fix: Render projects to only active users

### LINKED ISSUE #395: [BUG]:  render projects only with `access.status === 'active'`
### Describe the bug

We are currently rendering all projects regardless of `access.status` on `/projects` endpoint. 

### Steps to reproduce

1. Setup project
2. Run seed and login
3. Go to `/projects`
4. You will see all projects you have access to, regardless of your `access.status`

### Expected behavior

We need to do two things:-

1. Render projects on `/projects` with `access.status === 'active'`
2. Render 404 when users visit project's children endpoints (eg: `projects/{slug}/branches` if their `access.status` is not `active`

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ttebify**
# Description

> Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #395

## Type of change

> Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I documented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by dahal — CHANGES_REQUESTED** [review]
@ttebify#9024 thanks for submitting the PR, `withAccessControl` is amazing like we discussed.

There is one additional thing we need to fix, `/projects` page currently is rendering all the projects user has access, regardless of their status, let's restrict it to only render projects users have access to with status `active`.

**Review by chetannn — DISMISSED** [review]

**Review by dahal — CHANGES_REQUESTED** [review]
Looks like this dropdown needs fixing. `/projects` page is not rendering project without `access.active` status but this dropdown is.

image

File: apps/platform/pages/projects/index.tsx [hunk 1]
```diff
@@ -78,7 +77,9 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
       },
       include: {
         access: {
-          where: {},
+          where: {
+            status: MembershipStatus.active,
```
Comment (dahal) [inline apps/platform/pages/projects/index.tsx:81]: 👍🏼

**Comment by ttebify** [comment]
> Looks like this dropdown needs fixing. `/projects` page is not rendering project without `access.active` status but this dropdown is.
> 
> image

That was a good catch. I have resolved it now.

File: apps/platform/prisma/seeds/access.ts [hunk 2]
```diff
@@ -39,7 +39,9 @@ const seedAccesses = async () => {
             ]) as AccessType["role"]);
 
       const status =
-        projectOwner[project.id] === user.id ? "active" : "inactive";
+        projectOwner[project.id] === user.id
```
Comment (dahal) [inline apps/platform/prisma/seeds/access.ts:42]: 👍🏼

**Pull request merged**